LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -490,5 +551,142 @@ class PackageEditorExperimentsO {
+                return;
+            }
+            using IMEPackage baldPcc = MEPackageHandler.OpenMEPackage(baldPccPath);
+
+            ExportEntry baldMorph = baldPcc.FindExport(game.IsGame3() ? "BioChar_CitHub.Faces.HMM_Deco_1" :
+                game.IsGame2() ? "BIOG_Hench_FAC.HMM.hench_wilson" : "BIOA_UNC_FAC.HMM.Plot.FRE32_BioticLeader");
+            if (baldMorph == null || baldMorph.ClassName != "BioMorphFace")
+            {
+                ShowError($"Could not find the bald headmorph. Please ensure the vanilla game files have not been modified.");
+                return;
+            }
+
+            ExportEntry targetMorph = (ExportEntry)pew.SelectedItem.Entry;
+
+            BioMorphFace baldMorphFace = ObjectBinary.From<BioMorphFace>(baldMorph);
+            BioMorphFace targetMorphFace = ObjectBinary.From<BioMorphFace>(targetMorph);
+
+            if (baldMorphFace.LODs[0].Length != targetMorphFace.LODs[0].Length)
+            {
+                ShowError($"The selected headmorph differs from the expected one. This experiment only works for male human headmorphs.");
+                return;
+            }
+
+            List<string> targetNames = new List<string>()
+                {
+                    "Afro",
+                    "Buzzcut",
+                    "BuzzCut_WidowsPeak",
+                    "Deiter",
+                    "Greezer",
+                    "HAIR_pulledbackslick",
+                    "HAIR_sidepart",
+                    "HAIR_slickWidowsPeak",
+                    "HAIR_pulledBackBig",
+                    "Hair_splitSide",
+                    "HAIR_centerPart",
+                    "Flattop",
+                    "flattop_widowspeak",
+                    "Eastwood",
+                    "widowsPeak",
+                    "rollins",
+                    "straightHairLine",
+                };
+
+            // Get a list of the hair morph targets that exist in the targetMorph morphFeatures property.
+            // If the property does not exist or no feature matches the targetNames, we'll use all the default hair targets.
+            List<string> targetsInMorph = targetNames.Where(targetName =>
+            {
+                ArrayProperty<StructProperty> features = targetMorph.GetProperty<ArrayProperty<StructProperty>>("m_aMorphFeatures");
+                if (features.Any())
+                {
+                        // Compare the targetName against the names of the features
+                        List<string> featureNames = features.Select(feature => feature.GetProp<NameProperty>("sFeatureName").Value.Name).ToList();
+                    return featureNames.Exists(featureName => string.Equals(featureName, targetName, StringComparison.OrdinalIgnoreCase));
+                }
+                else { return true; }
+            }).ToList();
```

Comment:
Not asking you to change this, but as a suggestion for the future:
Remember that `Where` is a loop, and that everything inside the lambda will be executed on every iteration. When you have a calculation that does not depend on the loop variable (`targetName` in this case), you want to lift that out of the loop so it only gets calculated once. For a small dataset like this, inefficient code won't have a noticable impact, but it's good to get into the habit of writing efficient code (which is often easier to understand too).

```suggestion
            var features = targetMorph.GetProperty<ArrayProperty<StructProperty>>("m_aMorphFeatures");
            List<string> targetsInMorph;
            if (features.Any())
            {
                var featureNames = features.Select(feature => feature.GetProp<NameProperty>("sFeatureName").Value.Name).ToList();
                //Intersect finds the items the two enumerables have in common
                targetsInMorph = targetNames.Intersect(featureNames, StringComparer.OrdinalIgnoreCase).ToList();
            }
            else
            {
                targetsInMorph = targetNames;
            }
```